garak/configurable.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import inspect
+from garak import _config
+from garak import _plugins
+
+
+class Configurable:
+    def _load_config(self, config_root=_config):
+        if hasattr(self, "loaded"):
+            return  # only load once, this will ensure the config is not rerun for extending classes
+        local_root = (
+            config_root.plugins if hasattr(config_root, "plugins") else config_root
```

Comment:
This is checking if we have the existing global `_config` object although it could just test the instance type, there are a few different ways we could test, open to other preferences. I do think this may change as work progresses on solving for #646.

Reply:
In the longer term I see the top-level "plugins" entry disappearing and being replaced by its children as new top-level objects, I guess another kind of check here might avoid a bug if/when we get around to that change